src/pages/index.js

```diff
@@ -0,0 +1,301 @@
+const newPostModal = document.querySelector("#post-modal");
+const newPostForm = newPostModal.querySelector(".modal__form");
+const newPostCloseButton = newPostModal.querySelector(".modal__close-button");
+const newPostlinkInput = newPostModal.querySelector("#image-link-input");
+const newPostCaption = newPostModal.querySelector("#image-caption-input");
+const closeButtons = document.querySelectorAll(".modal__close-button");
+const postSubmitButton = newPostModal.querySelector(".modal__save-button");
+
+const editButton = document.querySelector(".profile__edit-button");
+const addButton = document.querySelector(".profile__add-button");
+
+//new instance
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "a7e2f32d-b266-4245-9f55-f8910f712974",
+    "Content-Type": "application/json",
+  },
+});
+
+api
+  .getAppInfo()
+  .then(([data, userData]) => {
+    console.log("Initial card data:", data);
+    api._userId = userData._id;
+
+    profileAvatar.src = userData.avatar;
+    profileName.textContent = userData.name;
+    profileDescription.textContent = userData.about;
+
+    data.forEach((item) => {
+      renderCard(item);
+    });
+  })
+
+  .catch((err) => {
+    console.error(err);
+  });
+
+//functions
+function renderCard(item, method = "prepend") {
+  const cardElement = getCardElement(item);
+  cardsList[method](cardElement);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown", handleEscapeKey);
+  modal.removeEventListener("mousedown", handleOutsideClick);
+}
+
+function getCardElement(data) {
+  const cardElement = cardTemplate.content
+    .querySelector(".card")
+    .cloneNode(true);
+  const cardNameElement = cardElement.querySelector(".card__text");
+  const cardImage = cardElement.querySelector(".card__image");
+  const cardLikeButton = cardElement.querySelector(".card__like-button");
+  const cardDeleteButton = cardElement.querySelector(".card__delete-button");
+
+  cardNameElement.textContent = data.name;
+  cardImage.src = data.link;
+  cardImage.alt = data.name;
+
+  if (data.isLiked) {
+    cardLikeButton.classList.add("card__like-button_liked");
+  }
+
+  cardDeleteButton.addEventListener("click", () =>
+    handleDeleteCard(cardElement, data._id)
+  );
+
+  deleteModalCancelBtn.addEventListener("click", () => {
+    closeModal(deleteModal);
+  });
+
+  cardLikeButton.addEventListener("click", (evt) => {
+    handleLike(evt, data._id);
+  });
+
+  function handleLike(evt, id) {
+    let isLiked = evt.target.classList.contains("card__like-button_liked");
+
+    api
+      .handleLikeStatus(id, isLiked)
+      .then((res) => {
+        if (res) {
+          evt.target.classList.toggle("card__like-button_liked");
+          const likesCount = evt.target.querySelector(".card__like-count");
+          if (likesCount) {
+            likesCount.textContent = res.likes.length;
```

Comment:
You don't have a `counter`. Please, delete it


Your code will be very simple here:

```js
  function handleLike(evt, id) {
    const isLiked = evt.target.classList.contains("card__like-button_liked");

    api
      .handleLikeStatus(id, isLiked)
      .then(() => {
        evt.target.classList.toggle("card__like-button_liked");
      })
      .catch(console.error);
  }
```